### DISCUSSION TITLE
how to change backup directory path once submitted?

### DISCUSSION THREAD

**Discussion by plrnr** [nextcloud/all-in-one#596] [Questions] [answered]
how to change backup directory path once submitted? I just tried to get a feel for it and put in a path. But now I want to change it. How can I do this?

**Comment by szaimen** [comment]
Hello, unfortunately there is no simple way to do this. The most easiest one is probably restoring the instance from the backup archive. Then you also have the option to choose a different path.

So basically:
1. create a backup
2. only continue if it is successful
3. save the backup password at a safe place
4. shut down the mastercontainer
5. prune all containers and volumes (see https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance)
6. move the backup archive to a different place
7. Install AIO again and don't anter a domain in the first step! Instead enter in the first step in the instance restore section the new path and the backup password
9. done

**Reply by 4y55** [reply]
Unfortunately it doesn't work for me. I chose a wrong path and the backup failed, now I need to reset to make a valid backup. Isn't there an option to change it like there is with the timezone?

**Reply by szaimen** [reply]
The logic is already implemented in the frontend if the first backup fails. 
However the backend was missing which gets implemented with https://github.com/nextcloud/all-in-one/pull/1107

**Reply by 4y55** [reply]
Can I currently edit manually at some point or should I wait for updates with the function?

**Reply by szaimen** [reply]
I would probably recommend to wait but since the update will take a while until it is avaiable, following https://github.com/nextcloud/all-in-one/discussions/1053 wil have you covered.

**Reply by szaimen** [reply]
This is now released with v2.0.2 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by AmIBeingObtuse** [reply]
Hi, could you advise the step by step instruction for changing the path of the backup location. You mentioned this is available in version 2.0.2. I am now running 2.0.3 but cannot see an option to change the location of the backup. I accidentally put the location as Backups get created in the following directory on the host: /mnt/nextcloud_aio_mastercontainer/borg/borg instead of nextcloud_aio_backupdir stupid copy and paste thing! Love the whole all in one project! this is my only issue.

**Reply by szaimen** [reply]
Exact steps are here: https://github.com/nextcloud/all-in-one/discussions/596#discussioncomment-2708493

However if the first backup failed, it should offer you to change the path again.

**Reply by claudsonm** [reply]
Would be good to make this process easier. The good thing is that I had to test the whole recovery flow in order to change the folder, so if I have to do it, I already know how it goes. haha

**Reply by chiva** [reply]
@szaimen I did follow all the steps 1 to 9 to reposition my backup path, they all went smooth.
But went to do a new backup after restoring, and now having this error message:

```
Cannot initialize a new repository as that was already done at least one time.
If you still want to do so, you may delete the ‘borg.config’ file that is stored in the mastercontainer volume manually, which will allow you to initialize a new borg repository in the chosen directory.
```

Is it safe to do so?
Would anything change on the borg repo?
Would the backup key still be the same?
Would it make a diff backup using the previous files or would a new one be made?
Any way I could continue using the existing repo?

**Reply by chiva** [reply]
I'm a bit blocked on this

Ended up deleting the `borg.config` file.
New backups were correctly done on-demand, but the Backup menu on AIO where you could change the daily time and Verify backups disappeared.

Ended up resetting everything and start from step 1 again.
New AIO started, set domain, select the Addons, set timezone...
Set also the backup path, confirmed it, and no option to Restore! (it also didn't ask for the recovery password)

And now it's showing a different backup password on the interface!

I really don't know how to continue from here, and starting not to trust the backup/recovery process 😣 
@szaimen can you give some guidance?

This is the screenshot right after adding the backup path where my backup lives.
It also looked exactly the same before resetting it (after I restored the instance to change the backup path by following the steps on @szaimen message). No way to verify, nor showing the different backup snapshots, or option to change the hour on when to make the automatic backup.

It looks as if I never made the first backup.

![CleanShot 2023-05-03 at 22 13 53](https://user-images.githubusercontent.com/305333/236040022-12c55fe9-85ee-48e7-98c6-cd55a0bb93d6.png)

**Reply by szaimen** [reply]
You apparently missed to enter the backup path and backup password instead of a domain during the first step on a clean aio instance. This would have restord the instance from backup.

**Reply by szaimen** [reply]
I adjusted the steps a bit to make things hopefully clearer

**Reply by chiva** [reply]
You're right!
Did add the domain and it removed the option to restore.
So... reset again everything.

Now correctly restored the backup as per your steps.

The backup section appears like this:
![CleanShot 2023-05-05 at 00 38 53](https://user-images.githubusercontent.com/305333/236344669-9c2916eb-8edf-4488-9eb9-b491eba0f45b.png)

@szaimen is this expected?
It doesn't give the option to verify existing backups, nor to configure the time for daily automatic backups.

**Reply by chiva** [reply]
I've just tested starting the containers and the backup section looks exactly the same, no way to change the options nor verify

**Reply by chiva** [reply]
Ended up pushing `Create backup`
On the logs I saw it said something regarding "compacting archive indices" or something like that, then it said "Done", but load on the container was still high, so it was doing still something.
It was still like that for a few hours, and went to sleep.

Then I went now to check on the logs this morning and saw it has done another backup?
Seems to be another run of the backup process, as there's no trace of the logs of the 1st backup I did after restoring (check paragraph above)

Based on the logs, it looks like it started automatically at ~5AM?
Which is strange, if it recovered the old config, I had it set to do automatic backups at 23h.
```
Creating archive at "/mnt/borgbackup/borg::20230505_051838-nextcloud-aio"
------------------------------------------------------------------------------
Repository: /mnt/borgbackup/borg
Archive name: 20230505_051838-nextcloud-aio
Archive fingerprint: xxxxxxxxxxxxx
Time (start): Fri, 2023-05-05 05:18:39
Time (end):   Fri, 2023-05-05 05:31:56
Duration: 13 minutes 17.48 seconds
Number of files: 906244
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:                1.98 TB              1.94 TB              1.98 MB
All archives:               17.72 TB             17.34 TB              1.50 TB

                       Unique chunks         Total chunks
Chunk index:                 1242061             14085141
------------------------------------------------------------------------------
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
Deleted data:                    0 B                  0 B                  0 B
All archives:               17.72 TB             17.34 TB              1.50 TB

                       Unique chunks         Total chunks
Chunk index:                 1242061             14085141
------------------------------------------------------------------------------
Performing backup...
Starting the backup...
Pruning the archives...
Compacting the archives...
Backup finished successfully on 05.05.2023 - 05:32:25 (00 hours 13 minutes 47 seconds)
```

The backup section on AIO, still looks exactly the same as above.
No option to verify a backup nor to change the backup automatic start time.
![image](https://user-images.githubusercontent.com/305333/236416436-70b2d66f-795e-4b88-aa04-9a5388750d6b.png)

Seems the backup options are not correctly restored and leaves it in a borked state 😬

**Reply by szaimen** [reply]
I think the whole situation comes from the fact that you removed the borg.config file (and possibly other stuff) in the data directory on the former instance, created a backup and restored that half-broken state on the clean aio-instance. You should be able to make the options visible again by copying the file manually from the borg archive to the data directory inside the mastercontainer.

I guess we will need to add a check in the future to prevent creating a new backup if the borg.config file is not found at the correct place.

**Reply by chiva** [reply]
Alright, have restored `borg.config` from the backups into its original path and the options are back.
Daily backup is being made at the expected time.
And no errors when creating a new backup 🙌 

But I seriously don't know why on the 1st restore I did, I was getting this error:
```
Cannot initialize a new repository as that was already done at least one time.
If you still want to do so, you may delete the ‘borg.config’ file that is stored in the mastercontainer volume manually, which will allow you to initialize a new borg repository in the chosen directory.
```
When I followed the steps correctly that time.

But well, now it's working.
Thanks @szaimen 🙇

**Reply by chiva** [reply]
@szaimen hit again the same error as on the 1st restore 😬
Didn't change anything since the last successful backup from last message.

Events since I first installed AIO are as follows:

- Installed AIO from scratch, but set wrong backup path.
- On the 1st restore I didn't delete anything and hit the error below.
- Then, deleted the config file, created a new snapshot/backup, and restored it
- On the 2nd restore (current one) I couldn't see the backup options due to the missing config file.
- Restored config file from the first snapshot as you indicated
- Did a backup correctly 3 days ago
- Now went to do a manual one, and hit the same error again.

```
Cannot initialize a new repository as that was already done at least one time.
If you still want to do so, you may delete the 'borg.config' file that is stored in the mastercontainer volume manually, which will allow you to initialize a new borg repository in the chosen directory:
sudo docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config
```

How should proceed here?

**Reply by szaimen** [reply]
> How should proceed here?

Is the drive still mounted where the backup archive is located?

**Reply by chiva** [reply]
Thanks again @szaimen ! That was it
Could we improve the error message to something like:

```
The folder doesn't contain a borg backup respository.
Cannot initialize a new repository as that was already done at least one time.
If you still want to do so, you may delete the 'borg.config' file that is stored in the mastercontainer volume manually, which will allow you to initialize a new borg repository in the chosen directory:
sudo docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config
```

That way it should be more clear for users why it's failing 😄

**Reply by szaimen** [reply]
https://github.com/nextcloud/all-in-one/pull/2520

**Reply by funkybside** [reply]
> Hello, unfortunately there is no simple way to do this. The most easiest one is probably restoring the instance from the backup archive. Then you also have the option to choose a different path.
> 
> So basically:
> 
> 1. create a backup
> 2. only continue if it is successful
> 3. save the backup password at a safe place
> 4. shut down the mastercontainer
> 5. prune all containers and volumes (see https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance)
> 6. move the backup archive to a different place
> 7. Install AIO again and don't anter a domain in the first step! Instead enter in the first step in the instance restore section the new path and the backup password
> 8. done

imo that process is absurd just to change a backup destination.   I hope making this a simple path edit in the AIO's webui is on the docket.   I want to change my destination simply because the aio results in root owner on unraid, which prevents duplicity and duplicati from successfully reading the files created (and I am not okay with running either of those as root as a workaround).   I wasn't aware the AIO would do this, I stored the backups in a folder I know was already covered by my remote backup solution so now it's either basically reinstall NC AIO ro deal with the warnings (and failed remote backups) this results in.

**Comment by plrnr** [comment]
thanks I will try that. After restore, is it also going to restore the backup path according to what it was when the backup was taken?

**Reply by szaimen** [reply]
No, it will use the new backup path that was chosen for the restore.

**Reply by plrnr** [reply]
thanks!

**Comment by jzcad1828** [comment]
I ran across the same issue, I submitted realizing I submitted the wrong path. Instead of recreating everything, I stopped the containers from the AIO page and modified the "borg_backup_host_location" property in the ../nextcloud_aio_mastercontainer/_data/data/configuration.json file on the host. As soon as I refreshed the AIO page I could see the updated backup path, and taking a backup put the backups in the updated location.

EDIT: I should mention that I made this change BEFORE I created an initial backup, so borg hadn't yet created the initial repo in the wrong location. Still, I imagine you should be able to move the data from the incorrect location to the correct location and update the config to both see the old backups and store the new ones there, but I haven't tested that.

**Reply by marcgror** [reply]
Thanks, working great

**Reply by Maximilian-Staab** [reply]
I did this after I already initialized the backup. Used this to switch from a mount point on the host to a docker nfs volume.

**Reply by BraianPita** [reply]
Thanks for this information. I was able to successfully move my borg folder from my old to my new backups location and change the configuration file as you mention here. Then, I restarted my mastercontainer and everything worked fine and I was able to create a new increment of my backup without issues

**Reply by thiswillbeyourgithub** [reply]
For anyone still struggling with this in 2025: The `configuration.json` file is in `/mnt/docker-aio-config/data/configuration.json` inside the `nextcloud-aio-mastercontainer` container.

**Comment by rb-26** [comment]
Thanks @jzcad1828 for your notes! 

My example was that I had set the location then had already run the initial backup. Here is what I did:
- Stopped the containers from the AIO interface
- Modified the "borg_backup_host_location" property in /nextcloud_aio_mastercontainer/data/configuration.json file
- Tried to backup and got an error with a command to run to resolve it
- Ran "docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config"
- Ran initial backup again from AIO interface
- Backup completed to new location with no issues

Hope this helps someone in the future who might be in the same boat as I was.

**Reply by wiwixu** [reply]
This work 100% to me on Nextcloud version (31.0.5) Thanks

**Comment by D3monX** [comment]
Confirm that deleting borg.config and modifing configuration.json seem working good even if first backup was already succesfull.

So i did:

1) shell inside the mastercontainer:
`docker exec -it nextcloud-aio-mastercontainer /bin/bash `
2) cd into the configuration dir:
`cd /mnt/docker-aio-config/data/`
3) delete o rename borg.config
`rm borg.config`
4) modify  the value of "borg_backup_host_location" in configuration.json
`vi configuration.json` (nano not present inside the container so needed to use vi)
5) exit from the nextcloud-aio-mastercontainer
`exit`

At this point, in the AIO web interface it say the first backup was failed, and i was able to set new location of backups and make a new one.

Would be nice if devs would add this functionality on the AIO web frontend. It should be just a script to change the value in the configuration.json and delete the borg.config (maybe with a option to clean/delete old backup directory)

**Reply by EddyXorb** [reply]
I confirm it worked.

**Reply by Weidav** [reply]
I can also confirm. You can leave the "borg_backup_host_location" empty and it let's you chose a new one in the in AIO web interface. I was unsure if I had to choose a correct one in the "config.json".

**Reply by cya9nide** [reply]
Works.  Found this searching for answer.  Thanks!

**Reply by ajmaradiaga** [reply]
This works perfectly!

**Reply by git529** [reply]
This worked for me too. Only difference is that I had to `mv borg.config borg.config.bak` then `vi configuratio.joson` then had to `mv borg.config.bak borg.config` again, otherwise the backup would fail.

**Reply by DarkLordGMS** [reply]
You no longer have to do this. You can just run this command and the "Reset backup location" button will show up in the AIO Interface

sudo docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config

**Reply by funkybside** [reply]
Thank you, and 100% hard agree.  It's bonkers that the 'official' solution is basically a reinstall of the system.

**Comment by DarkLordGMS** [comment]
To reset the backup location you just have run this command and the "Reset backup location" button will show up in the AIO Interface

sudo docker exec nextcloud-aio-mastercontainer rm /mnt/docker-aio-config/data/borg.config

**Reply by Eysenor** [reply]
This is the way. Did it yesterday and it worked super easily with AIO on unraid. 
Why can't the botton just be there all the time I do not understand. Just clicked the botton, chose the new location and remade the backup. I tried before moving the old backups to the new location but it did not work making a backup when those where there. 
So just point to an empty folder and redo the backup.

Thanks a lot for the advice, this should be the one answer marked as solution.

**Comment by mbc0** [comment]
Hi,

Hoping there is someone here that understands how this works on unraid as I cannot seem to find a way to set a backup path outside of the docker file?  I have set user files to an unraid share which does seem to work for that purpose but how can I store the backups outside of the docker?  Thank you (hoping!!)

From NextcloudAIO 
Backups will be created in the following directory on the host: /NEXTCLOUD_DATADIR/backups/borg

![image](https://github.com/user-attachments/assets/ecea8474-a6e6-4bf6-a8a8-762d49a96db4)

The unraid template
![image](https://github.com/user-attachments/assets/af6ba3cd-a979-406f-a928-1faa11c9d83c)

**Reply by chiva** [reply]
From the looks of your terminal, backups are being stored on the host, outside of the Docker container.
And regarding the path `/NEXTCLOUD_DATADIR/`, seems like you set literally that as the backup path.
You have to define the path from the root of your host machine (Unraid), it is not looking into Docker.

You will need to change the backup path from `/NEXTCLOUD_DATADIR/` (which I'm not sure it'd even survive a reboot on Unraid) to something like `/mnt/user/appdata/nextcloudaio/borg_backup`, or create a backup Share for example `/mnt/user/nextcloud_backup/` or point it to an NFS/Samba share, really depends on your setup.

**Comment by mbc0** [comment]
Thank you so much for your response! 

in the docker template there is a variable (not path) for NEXTCLOUD_DATADIR and it is set to /mnt/user/nextcloudaio  This created a share on my unraid server which I can access and it is definitely on my shares
![image](https://github.com/user-attachments/assets/766b4d76-9523-4266-8071-25ece1df2f03)

I can see the files are on disk 14 on my server
![image](https://github.com/user-attachments/assets/5fe8b14f-96f8-41e5-a316-954a19338af2)

What I would like to do is change the backup path to that location and create a backup folder or even better change the location to my backup folder on unraid (/mnt/user/Backups) However I have 2 issues

1, how do I access the nextcloudaio share from nextcloudaio? I cannot find the path?
2, how can I add a path to my mnt/user/Backups folder in the template? I have tried adding paths but I cannot access them? should I been adding paths as a variable? (this does not make sense to me but must be working as the variable NEXTCLOUD_DATADIR/ seems to connect to my shares?

I appreciate all your help! thank you

**Reply by chiva** [reply]
Hey @mbc0 !

Best if we can keep the conversation within a thread, by answering to the last message, so we don't spawn different convesations 😉 

1. Looks like everything is fine then. If the share exists and the path is `NEXTCLOUD_DATADIR` environment variable of the Docker template (your 1st image), then you don't need anything else, all good.
I don't think `NEXTCLOUD_DATADIR` is used in the AIO container directly.
It'll be passed on and used by a child Nextcloud container that it spawns when starting it, as it launches 4-6 containers (depending on your feature selection), but you don't really need to know where is it used internally (I don't, that's the good thing of the AIO).

2. The backup path is not set in the template, it just needs to be defined in the AIO interface
So, start the AIO container, visit the admin page:
```https://<YOUR_AIO_IP:8089/containers```
Go to the `Backup and restore` section, and you'll see a field to input the path for backups (it is not visible in my case as it'll disappear as soon as you set one).
Use the path of your host server (Unraid) for where you want the backup to be stored. AIO will then create a `backup` folder inside that path and store all the resulting Borg Backup files.
Recommended to use a different Share than the one for Appdata, so they are not mixed, something like a Share named `nextcloud_backup` and then use the path to that `/mnt/user/nextcloud_backup`.
And hopefully eventually setup something that copies it to a different server so you have proper backup redundancy 👍 .
![CleanShot 2024-07-31 at 16 37 31](https://github.com/user-attachments/assets/44e31887-3b8c-415a-bbe8-0005fe0140b5)

**Reply by mbc0** [reply]
So sorry, I did not mean to create a new thread! I have just hit reply to type this, so hope it is ok! 

I think I was overthinking/not thinking correctly, I just followed the fresh install/reset procedure and typed /mnt/user/Backups/nexcloudaio/  and run a backup to find everything in my shares! so great news & thank you so much for your time! all appears to be working well.

**Comment by SnejPro** [comment]
### How I switched from local storage to ssh storage

I wanted to switch to ssh repository without recreating the backup repository.

## Prerequisites:
1. The backup repository has been moved to ssh storage.
2. You should make a copy of the backup and store it somewhere else in case something goes horribly wrong.

## To-do:

1. SSH: Connect to the Docker host via SSH
3. SSH; shell inside mastercontainer
`docker exec -it nextcloud-aio-mastercontainer /bin/bash`
4. SSH; change directory
`cd /mnt/docker-aio-config/data`
5. SSH: move config file temporarily
`mv borg.config borg.config.bkp`
6. Browser: open AIO frontend
7. Browser: scroll down to "reset location"
8. Browser: set new location path to borg repository: `ssh://[...]`
9. Browser: Click "create backup" (it will fail, because ssh-key is not set yet)
10. SSH: Move the configuration file back
`mv borg.config.bkp borg.config`
11. Browser: Get ssh-key from AIO 
12. Borg-storage: add ssh-key to ssh-host

Thanks to you all. :)

**Reply by burajiruhito** [reply]
This works like a charm. Thaks.

**Comment by szaimen** [comment, marked as answer]
Hi, just to mention it, we now show a `Reset backup location` button all the time: https://github.com/nextcloud/all-in-one/pull/7185

This is now released with v12.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
